service/src/main/java/bio/terra/externalcreds/models/FenceAccountKey.java

```diff
@@ -0,0 +1,18 @@
+package bio.terra.externalcreds.models;
+
+import java.sql.Timestamp;
+import java.util.Optional;
+import org.immutables.value.Value;
+
+@Value.Immutable
+public interface FenceAccountKey extends WithFenceAccountKey {
+  Optional<Integer> getId();
```

Comment:
FYI this is optional because when we pass a `FenceAccountKey` object to `insertFenceAccountKey`, it won't have an id at that point.